core/offchain/src/api.rs

```diff
@@ -188,7 +190,19 @@ impl<Storage, KeyProvider> OffchainExt for Api<Storage, KeyProvider> where
 	}
 
 	fn timestamp(&mut self) -> Timestamp {
-		unavailable_yet("timestamp")
+		let now = SystemTime::now();
+		let epoch_duration = now.duration_since(SystemTime::UNIX_EPOCH);
+		match epoch_duration {
+			Err(_) => {
+				warn!("Current time is earlier than UNIX_EPOCH.");
```

Comment:
It is okay to panic in this case, but as this should only affect horribly misconfigured Windows systems, it does not matter.

Reply:
I'd rather keep it as non panicker and just default to `0` without even printing a warning.